File: JavaScript/test/electric_rose_test.js
var expect = require('chai').expect;
var rose = require('../src/electric_rose.js');

describe("Electric Rose", function() {

  it("should foo", function() {
    var batteries = [new rose.Battery("foo", false, 1)];
    var app = new rose.ElectricRose(batteries);
    app.updateCharge();
    expect(batteries[0].type).to.equal("fixme");
  });

});

File: Go/electric-rose_test.go
package main

import "testing"

func Test_Foo(t *testing.T) {
	var batteries = []*Battery{
		{"foo", false, 1},
	}

	UpdateCharge(batteries)

	if batteries[0].brand != "fixme" {
		t.Errorf("Brand: Expected %s but got %s ", "fixme", batteries[0].brand)
	}
}
